lib/Backends/BackendUtils.cpp

```diff
@@ -81,6 +74,38 @@ runtime::RuntimeBundle::getSymbolInfo(const Named *v) const {
   return it->second;
 }
 
+/// If \p PH is an output placeholder, \returns true.
+/// This is determined by checking if the PH has a user which uses the PH as an
+/// overwritten input.
+bool checkIfOutput(const Placeholder *PH) {
+  for (auto &use : PH->getUsers()) {
+    // Look through the inputs of the PH's users. If an input is overwritten
+    // check if it's the PH, if it is return true.
+    for (unsigned i = 0; i < use.getUser()->getNumInputs(); i++) {
+      if (use.getUser()->isOverwrittenNthInput(i)) {
```

Comment:
IMO this is a good place to use `continue` instead of deeply nesting the control flow.

Reply:
http://www.llvm.org/docs/CodingStandards.html#use-early-exits-and-continue-to-simplify-code

Reply:
I'm not sure I understand, Don't I need to examine all overwritten inputs to make sure the PH is not an output? I'm not seeing where I can short circuit the loop.